File: OWASP-ZAP-Scripts/download-js-files.js
// passive script to download js files of desired host while navigating through the app

var PluginPassiveScanner = Java.type("org.zaproxy.zap.extension.pscan.PluginPassiveScanner");
var FileUtils = Java.type("org.apache.commons.io.FileUtils");
var Desktop = Java.type("java.awt.Desktop");
var File = Java.type("java.io.File");
var System = Java.type("java.lang.System");

function scan(ps, msg, src) {
    // Check if the message is a response containing JavaScript
    var contentType = msg.getResponseHeader().getHeader("Content-Type");
    if (contentType != null && contentType.contains("javascript") 
	   && msg.getRequestHeader().getHostName().toString().match(/^.*?google\..*$/)) {
        // Save the JavaScript file
        var fileName = msg.getRequestHeader().getURI().getPath().replace(/\//g, "_");
        var response = msg.getResponseBody().toString();
        try {
		  var zapDir = new File(System.getProperty("user.home") + "/Desktop/<YOUR>/<DESIRED>/<PATH>");
            zapDir.mkdirs();
            FileUtils.writeStringToFile(new File(zapDir, fileName), response, "UTF-8");
		  print("File saved: " + fileName + "\nFrom Host: " + msg.getRequestHeader().getHostName())
        } catch (e) {
            print(e)
        }
    }
}

function appliesToHistoryType(historyType) {
    return PluginPassiveScanner.getDefaultHistoryTypes().contains(historyType);
}

File: OWASP-ZAP-Scripts/download-js-files.py
from org.zaproxy.zap.extension.pscan import PluginPassiveScanner
from org.apache.commons.io import FileUtils
from java.awt import Desktop
from java.io import File
from java.lang import System

def scan(ps, msg, src):
    # Check if the message is a response containing JavaScript
    content_type = msg.getResponseHeader().getHeader("Content-Type")
    if content_type is not None and "javascript" in content_type.lower() and \
        msg.getRequestHeader().getHostName().toString().match(r'^.*?google\..*$'):
        # Save the JavaScript file
        file_name = msg.getRequestHeader().getURI().getPath().replace('/', '_')
        response = msg.getResponseBody().toString()
        try:
            zap_dir = File(System.getProperty("user.home") + "/Desktop/<YOUR>/<DESIRED>/<PATH>")
            zap_dir.mkdirs()
            FileUtils.writeStringToFile(File(zap_dir, file_name), response, "UTF-8")
            print("File saved: " + file_name + "\nFrom Host: " + msg.getRequestHeader().getHostName())
        except Exception as e:
            print(e)

def appliesToHistoryType(history_type):
    return PluginPassiveScanner.getDefaultHistoryTypes().contains(history_type)
